### PR TITLE
SDL 0280 Revisions - Adding new parameter of requires audio support and bluetoothDeviceAddress

### PR DESCRIPTION
**Pull request by Akihiro-Miyazaki**
## Introduction
This is an update of SDL-0280.
Change the words from "HU" to "head unit (HU)" and from "BT" to "Bluetooth (BT)".

## Proposed solution
Change the words from "RAI" to "RegisterAppInterface" .
Add a description of MediaStreamingStatus class (Requirement from SDLC).
Add a description that the HMI is responsible for preventing app activation as described in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/941#issuecomment-644911918) (Requirement from SDLC).
Delete unnecessary lines and blanks.

## Alternatives considered
Delete the alternative (Requirement from SDLC).

### CODE REVIEW

**Comment by Akihiro-Miyazaki** [comment]
Hi, @theresalech -san.
I took over #941 from @Kazuki-Sugimoto because he was off this project.
This documentation was revised according to [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/941#issuecomment-648864859).
Also, some words was changed and the unnecessary lines and blanks was deleted.
Could you please review my proposal?
Thank you.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 1]
```diff
@@ -45,119 +44,126 @@ If the SDL app or HU cannot use this function, it will perform the same operatio
...
+      <param name="cloudConnectionStatus" type="Common.CloudConnectionStatus" mandatory="false"></param>
++     <param name="requiresAudioSupport" type="Boolean" mandatory="false">
++         <description>Set whether or not this app requires the use of an audio streaming output device.</description>
++     </param>
+  </struct>
 ```
 
+### MediaStreamingStatus class
+
+The library supports the new version of the protocol specification. However the app needs to use `MediaStreamingStatus` class before registering if the IVI system uses the earlier version.
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:158]: Should this be re-worded, per [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/941#issuecomment-644850433):
```suggestion
If the Java Suite library supports the newer version of the protocol specification, but the IVI system is using a lower version, the app will still need to use the `MediaStreamingStatus` class before attempting to register.
```

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 2]
```diff
@@ -45,119 +44,126 @@ If the SDL app or HU cannot use this function, it will perform the same operatio
...
+With the changes of the flow, it is necessary to do refactoring the library to move the logic of `MediaStreamingStatus` class.
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:159]:
```suggestion
With the changes of the flow, it is necessary to do refactoring of the Java Suite library to move the logic of the `MediaStreamingStatus` class.
```

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 3]
```diff
@@ -45,119 +44,126 @@ If the SDL app or HU cannot use this function, it will perform the same operatio
...
+
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:160]: Based on [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/941#issuecomment-644850433), additional information about the refactoring and details about how to use the `MediaStreamingStatus` class should be added to the proposal.  Can you please add?

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 4]
```diff
@@ -45,119 +44,126 @@ If the SDL app or HU cannot use this function, it will perform the same operatio
...
 ### Launch the app
 
-By using this function, the HU is able to perform  BT automatic connection based on the `bluetoothDeviceAddress` which is notified when the app that requests audio (`requiresAudioSupport` is set to true) is launched. 
+By using this function, the HU is able to perform  BT automatic connection based on the `bluetoothDeviceAddress` which is notified when the app that requests audio (`requiresAudioSupport` is set to true) is launched.
 It can also display a message that prompts BT connection to the user.
 
+Note: While the HMI is connecting with the Bluetooth mobile device of the app that is set up `requiresAudioSupport=true`, preventions from activations of the app are needed.
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:166]: I think this should be re-worded to be more aligned with [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/941#issuecomment-644911918)
```suggestion
Note: The HMI will be responsible for preventing app activation while the HMI is in the process of connecting to the mobile device's BT for an app that has `requiresAudioSupport=true`. For example, the HMI could show a loading icon for that app, or display an error message to the user upon activation.
```

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 5]
```diff
@@ -168,52 +174,7 @@ Due to the complexity of the flow, the developer must do the implementation care
 This proposal requires a major version change.
 Since new parameters are added, Core, iOS, Java Suite, RPC, Protocol, and HMI are affected.
 
-
 ## Alternatives considered
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:177]: The Steering Committee's requested revisions did not include removing this section.  Even though we will be using the flow described in the `Proposed solution` section, the flow described here is still an alternative that was considered, so should be kept for visibility.  Please add back lines 174 - 219.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 6]
```diff
@@ -168,52 +174,7 @@ Due to the complexity of the flow, the developer must do the implementation care
...
 
-The flow in the Proposed solution section is complicated, so the author has also considered the simplified solution below.
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:174]: You can update this section to reflect the reason why we decided to move forward with the flow described in the `Proposed solution` section:
```suggestion
While this alternative was considered, the Steering Committee determined this flow does not account for previous versions of the protocol, and therefore is not viable as the primary solution.
```

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 7]
```diff
@@ -168,52 +174,7 @@ Due to the complexity of the flow, the developer must do the implementation care
...
-
-### Change the app registration flow
-
-![registeringApp_flow2](../assets/proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress/registeringApp_flow2.png)
+There is still room for consideration, but I recommend this proposed solution.
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:179]:
```suggestion
```
Remove in favor of suggested change to line 174.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 5 above)
Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:177]: Hi, @theresalech -san.
Thank you for your review.
I will sdd back lines 174-219 and make a commit it.
Could you please confirm it.
Thank you.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 6 above)
Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:174]: Hi, @theresalech -san.
Thank you for your review.
I will modify to this sentence after adding back lines 174-219 and make a commit it.
Could you please confirm it again.
Thank you.

**Comment by theresalech** [comment]
@Akihiro-Miyazaki, I've confirmed the your change to lines 174 - 219 and marked that comment as resolved.  Please advise once the final comment (https://github.com/smartdevicelink/sdl_evolution/pull/1043#discussion_r450258401) has been addressed and this PR is ready for my review.  Thank you!

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 6 above)
Comment (theresalech) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:174]: @Akihiro-Miyazaki, this does not yet appear to be resolved.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 5 above)
Comment (theresalech) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:177]: @Akihiro-Miyazaki, this does not yet appear to be resolved.

**Comment by theresalech** [comment]
@Akihiro-Miyazaki, I have responded to the questions you've left on some of my feedback.  Please advise once all feedback has been addressed and this PR is ready for me to review.  Thank you!

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 3 above)
Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:165]: Hi, @theresalech -san.
I am very sorry for delay my response.
I modified the proposal documentation and added the image file.
Also, I made a commit them.
Could you please confirm them again.
Thank you.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 5 above)
Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:182]: Hi, @theresalech -san.
I am very sorry for delay my response.
I modified the proposal documentation and added the image file.
Also, I made a commit them.
Could you please confirm them again.
Thank you.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 6 above)
Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:174]: Hi, @theresalech -san.
I am very sorry for delay my response.
I corressponded it and made a commit it.
Could you please confirm it again.
Thank you.

**Comment by Akihiro-Miyazaki** [comment]
I made a commit all your suggestion.
Could you please confirm them again.
Thank you.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 8]
```diff
@@ -156,7 +156,12 @@ HMI API:
 ### MediaStreamingStatus class
 
 If the Java Suite library supports the newer version of the protocol specification, but the IVI system is using a lower version, the app will still need to use the `MediaStreamingStatus` class before attempting to register.
-With the changes of the flow, it is necessary to do refactoring of the Java Suite library to move the logic of the `MediaStreamingStatus` class.
+By using the value of isAudioOutputAvailable() method on MediaStreamingStatus class, the Java Suite library can make a decision whether BT A2DP is connected like below:
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:159]:
```suggestion
By using the value of `isAudioOutputAvailable()` method on `MediaStreamingStatus` class, the Java Suite library can make a decision whether BT A2DP is connected like below:
```

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 9]
```diff
@@ -168,14 +179,13 @@ Due to the complexity of the flow, the developer must do the implementation care
...
-
 ## Alternatives considered
 
-The flow in the Proposed solution section is complicated, so the author has also considered the simplified solution below.
+While this alternative was considered, the Steering Committee determined this flow does not account for previous versions of the protocol, and therefore is not viable as the primary solution.
 
 ### Change the app registration flow
 
-![registeringApp_flow2](../assets/proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress/registeringApp_flow2.png)
+![registeringApp_flow1](../assets/proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress/registeringApp_flow1.png)
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:188]: This image is for the alternative solution, correct?  The Steering Committee didn't request any revisions to the alternative solution.  Can you please advise why you've changed this image?

Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:188]: Hi, @theresalech -san.
Thank you for your prompt reply.
Yes, it is correct.
I only changed the file name about the proposal documentation and image file.
Thank you.

Comment (theresalech) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:188]: @Akihiro-Miyazaki, I'm having trouble following if the flow depicted in the image is for the proposed solution or the solution described in the `Alternatives considered` section.  Regardless, this image change was not part of the Steering Committee's [agreed upon revisions](https://github.com/smartdevicelink/sdl_evolution/issues/941#issuecomment-648864859) and so should not be included in this PR.  Please revert to the image used previously, and if you would like a new image to be included in the proposal, you can comment on the review issue with the suggested change when the revised proposal is brought back into review.  Please let me know if you have any questions about this.  Thank you!

Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md]: @theresalech -san,
I modified a proposal documentation.
I also replaced the image files.
And I made a commit them.
Thank you.

**Comment by Akihiro-Miyazaki** [comment]
@theresalech -san.
Could you please confirm them again.
Thank you.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 9 above)
Comment (theresalech) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md]: @Akihiro-Miyazaki , the image file should **not** be replaced in this PR, as that was not requested by the Steering Committee.  Please revert to the image used previously (registeringApp_flow2.png).  Thank you.

Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md]: @theresalech -san,
Thank you for your response.
What I did only was to replace (revert) the image file to previous one.
If this comment was wrong, could you please teach me how to revert?
Thank you in advance for your help.

Comment (theresalech) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md]: @Akihiro-Miyazaki I've confirmed the new image is no longer present in the `Alternatives Considered` section.  Thanks!

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 10]
```diff
@@ -1,36 +1,35 @@
...
 However, with this proposal, by adding a new parameter, the SDL app always establishes the transport connection and is registered without depending on the connection status of BT A2DP. If BT A2DP is not connected, the HU will automatically connect BT using `bluetoothDeviceAddress` or request connection from the user.
 
-
 ### Change the app registration flow
 
 In order for the app to be always registered, the changes in SDL session establishment to app registration flow are shown below.
 If the SDL app or HU cannot use this function, it will perform the same operation as before.
 
-![registeringApp_flow](../assets/proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress/registeringApp_flow1.png)
+![registeringApp_flow](../assets/proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress/registeringApp_flow.png)
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:28]: The Steering Committee did not request this image change in the proposal revisions.  Please revert back to the original `registeringApp_Flow.png`.  If you would like to suggest updating this image, you can comment on the review issue with the suggested change when the revised proposal is brought back into review.  Thank you!

Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:28]: HI, @theresalech -san.
Thank you for your re-review.
I will replace (revert) the image file to previous one and make a commit it.
Could you please confirm it again.
Thank you.

**Comment by Akihiro-Miyazaki** [comment]
Hi, @theresalech -san.
I reverted the image file to previous one and made a commit it.
Could you please confirm it again.
Thank you.

**Pull request merged**